### DISCUSSION TITLE
How do I integrate User Control?

### DISCUSSION THREAD

**Discussion by Bodden007** [Live-Charts/LiveCharts2#1562] [Q&A] [answered]
Please show the full program code from this example:
https://livecharts.dev/docs/WPF/2.0.0-rc1/samples.general.realTime
With the MainWindow.xaml and MainWindow.xaml.cs classes. How do I integrate UserControl with them?

**Comment by beto-rodriguez** [comment, marked as answer]
Hello, the full code is in this repository:

https://github.com/beto-rodriguez/LiveCharts2/tree/master/samples

In your case you need the WPF sample.

[In the WPF sample](https://github.com/beto-rodriguez/LiveCharts2/tree/master/samples/WPFSample) you can find the main window, the important part is that the `<ContentControl></ContentControl>` tag is where the user control is displayed in the app.

But that should be irrelevant, you should be able to place the chart directly inside your window without any UserControl instance, a UserControl is not required to render a chart, just ignore that UserControl, it should work the same if you place the chart inside your MainWindow class.

If you want to learn more, you could clone this repo, and focus only in the WPF sample, try to run it and see how it works.

**Reply by Bodden007** [reply]
That's how it works
`using System.Text;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Data;
using System.Windows.Documents;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Imaging;
using System.Windows.Navigation;
using System.Windows.Shapes;

namespace WpfSample
{
    /// <summary>
    /// Interaction logic for MainWindow.xaml
    /// </summary>
    public partial class MainWindow : Window
    {
        public MainWindow()
        {
            InitializeComponent();
        }
    }
}`

`<Window x:Class="WpfSample.MainWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:local="clr-namespace:WpfSample"
        xmlns:myControl="clr-namespace:WpfSample;assembly=WpfSample" 
    xmlns:lvc="clr-namespace:LiveChartsCore.SkiaSharpView.WPF;assembly=LiveChartsCore.SkiaSharpView.WPF">


    <Window.DataContext>
        <local:ViewModel />
    </Window.DataContext>

    <lvc:CartesianChart
        Series="{Binding Series}">
    </lvc:CartesianChart>
</Window>
`
`using LiveChartsCore.SkiaSharpView;
using LiveChartsCore;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using CommunityToolkit.Mvvm.ComponentModel;
using LiveChartsCore.SkiaSharpView.Painting;
using SkiaSharp;
using LiveChartsCore.SkiaSharpView.VisualElements;

namespace WpfSample
{
    public class ViewModel
    {
         public ISeries[] Series { get; set; } 
            = new ISeries[]
            {
                new LineSeries<double>
                {
                    Values = new double[] { 2, 1, 3, 5, 3, 4, 6 },
                    Fill = null
                }
            };
    }
}
`

**Comment by Bodden007** [comment]
![Uploading 2024-08-07_201613.jpg…]()
